wapitiCore/definitions/ldapi.py

```diff
@@ -17,40 +17,62 @@
 # You should have received a copy of the GNU General Public License
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
-TYPE = "vulnerability"
-
-NAME = "LDAP Injection"
-SHORT_NAME = "LDAPI"
-
-WSTG_CODE = ["WSTG-INPV-06"]
-
-DESCRIPTION = (
-    "LDAP Injection is an attack used to exploit web based applications that construct LDAP statements based on "
-    "user input."
-) + " " + (
-    "When an application fails to properly sanitize user input, it’s possible to modify LDAP statements using "
-    "a local proxy."
-) + " " + (
-    "This could result in the execution of arbitrary commands such as granting permissions to unauthorized queries."
-)
-
-SOLUTION = (
-    "To protect against XPATH injection, Escape all variables using the right LDAP encoding function."
-) + " " + (
-    "Use Frameworks that Automatically Protect from LDAP Injection."
-)
-
-REFERENCES = [
-    {
-        "title": "LDAP Injection",
-        "url": "https://owasp.org/www-community/attacks/LDAP_Injection"
-    },
-    {
-        "title": "LDAP Injection Prevention Cheat Sheet",
-        "url": "https://cheatsheetseries.owasp.org/cheatsheets/LDAP_Injection_Prevention_Cheat_Sheet.html"
-    },
-    {
-        "title": "CWE-90: Improper Neutralization of Special Elements used in an LDAP Query ('LDAP Injection')",
-        "url": "https://cwe.mitre.org/data/definitions/90.html"
-    }
-]
+from typing import List
+
+from wapitiCore.definitions.base import FindingBase
+
+
+class LdapInjectionFinding(FindingBase):
+    @classmethod
+    def name(cls) -> str:
+        return "LDAP Injection"
+
+    @classmethod
+    def description(cls) -> str:
+        return (
+            "LDAP Injection is an attack used to exploit web based applications that construct LDAP statements "
+            "based on user input."
+        ) + " " + (
+            "When an application fails to properly sanitize user input, it’s possible to modify LDAP statements using "
+            "a local proxy."
+        ) + " " + (
+            "This could result in the execution of arbitrary commands such as granting permissions "
+            "to unauthorized queries."
+        )
+
```

Comment:
All these quotes and parentheses seem convoluted to me. Why don't you just write a regulare string ?

Reply:
Changed. It was a way to cut phrases for translations but it is not there anomore.